flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamJoin.scala

```diff
@@ -0,0 +1,241 @@
+import org.apache.flink.streaming.api.datastream.CoGroupedStreams.TaggedUnion
+import org.apache.flink.streaming.api.windowing.evictors.Evictor
+import org.apache.flink.streaming.api.windowing.evictors.Evictor.EvictorContext
+import java.lang.Iterable
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord
+import org.apache.flink.streaming.runtime.operators.windowing.TimestampedValue
+import org.apache.flink.api.common.functions.RichFlatJoinFunction
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.api.common.state.ValueState
+import org.apache.flink.api.common.state.ValueStateDescriptor
+import org.apache.flink.util.Collector
+
+class DataStreamJoin(
+  calc: LogicalJoin,
+  cluster: RelOptCluster,
+  traitSet: RelTraitSet,
+  inputLeft: RelNode,
+  inputRight: RelNode,
+  rowType: RelDataType,
+  description: String)
+    extends BiRel(cluster, traitSet, inputLeft, inputRight) with DataStreamRel {
+
+  override def deriveRowType(): RelDataType = rowType
+
+  override def copy(traitSet: RelTraitSet, inputs: java.util.List[RelNode]): RelNode = {
+    new DataStreamJoin(
+      calc,
+      cluster,
+      traitSet,
+      inputs.get(0),
+      inputs.get(1),
+      rowType,
+      description + calc.getId())
+  }
+
+  override def toString: String = {
+    s"Join(${
+      if (!calc.getCondition.isAlwaysTrue()) {
+        s"condition: (${calc.getCondition}), "
+      } else {
+        ""
+      }
+    }left: ($inputLeft), right($inputRight))"
+  }
+
+  override def explainTerms(pw: RelWriter): RelWriter = {
+    super.explainTerms(pw)
+      .itemIf("condition", calc.getCondition, !calc.getCondition.isAlwaysTrue())
+      .item("join", calc)
+      .item("left", inputLeft)
+      .item("right", inputRight)
+  }
+
+  override def translateToPlan(tableEnv: StreamTableEnvironment): DataStream[Row] = {
+
+    val inputDSLeft = inputLeft.asInstanceOf[DataStreamRel].translateToPlan(tableEnv)
+    val inputDSRight = inputRight.asInstanceOf[DataStreamRel].translateToPlan(tableEnv)
+
+    //define the setup for various types of joins to be supported
+    (calc.getCondition.isAlwaysTrue(), calc.getJoinType) match {
+      case (true, JoinRelType.LEFT) =>
+        createInnerQueryJoin(inputDSLeft, inputDSRight)
+      case (_, _) =>
+        throw new TableException("Table does not support this type of JOIN.")
+    }
+
+    null
+  }
+
+  def createInnerQueryJoin(
+    inputDSLeft: DataStream[Row], inputDSRight: DataStream[Row]): DataStream[Row] = {
+
+    // get the output types
+    val rowTypeInfo = FlinkTypeFactory.toInternalRowTypeInfo(getRowType).asInstanceOf[RowTypeInfo]
+
+    val result = inputDSLeft.join(inputDSRight)
+      .where(new EmptyKeySelector()).equalTo(new EmptyKeySelector())
+      .window(GlobalWindows.create())
+      .trigger(new ProcTimeLeftJoinTrigger())
+      .evictor(new FullEvictor())
+      .apply(new JoinProcTimeForInnerQuerry(rowTypeInfo))
+
+    null
```

Comment:
@shijinkui  thanks for the review...it is indeed redundant as i am catching actually all cases in the match pattern matching. I will remove it